torch/csrc/jit/script/compiler.cpp

```diff
@@ -1255,7 +1256,9 @@ struct to_ir {
             save_false->findInParentFrame(x)) {
           throw error;
         } else {
-          environment_stack->setVariableTypeError(x, error.what());
+          environment_stack->setVariableTypeError(x, [=]() -> std::string {
+            return error.what();
```

Comment:
little strange that we're lazily constructing the error message in the other case and eagerly constructing it here

Reply:
It's not on the hot path and it would need to capture a lot more stuff.